cmd/core/pdr.go

```diff
@@ -0,0 +1,116 @@
+package core
+
+import (
+	"fmt"
+	"github.com/edgecomllc/eupf/cmd/core/service"
+	"github.com/edgecomllc/eupf/cmd/ebpf"
+	"github.com/rs/zerolog/log"
+	"github.com/wmnsk/go-pfcp/ie"
+)
+
+func deletePDR(spdrInfo SPDRInfo, mapOperations ebpf.ForwardingPlaneController) error {
+	if spdrInfo.Ipv4 != nil {
+		if err := mapOperations.DeletePdrDownLink(spdrInfo.Ipv4); err != nil {
+			return fmt.Errorf("Can't delete IPv4 PDR: %s", err.Error())
+		}
+	} else if spdrInfo.Ipv6 != nil {
+		if err := mapOperations.DeleteDownlinkPdrIp6(spdrInfo.Ipv6); err != nil {
+			return fmt.Errorf("Can't delete IPv6 PDR: %s", err.Error())
+		}
+	} else {
+		if err := mapOperations.DeletePdrUpLink(spdrInfo.Teid); err != nil {
+			return fmt.Errorf("Can't delete GTP PDR: %s", err.Error())
+		}
+	}
+	return nil
+}
+
+func extractPDR(pdr *ie.IE, session *Session, spdrInfo *SPDRInfo, resourceManager *service.ResourceManager, teidCache map[uint8]uint32) error {
```

Comment:
Let's compose all necessary data for PDR creation in a separate struct(interface). 
This struct will consist of:

1. session ref for FARID/QERID allocation
2. global resource manager for FTEID/IP allocation
3. teid cache for choose id procedure

and provide corresponding interface methods:

1. Get FARID
2. GetQERID
3. GetFTEID
4. (GetUEIP)

Pass an object of this interface to `extractPDR` function to simplify it.